raiden/network/transport/matrix.py

```diff
@@ -266,19 +265,22 @@ def stop_and_wait(self):
         if not self._running:
             return
         self._running = False
-        self._client.set_presence_state(UserPresence.OFFLINE.value)
-        self._client.stop_listener_thread()
-        self._client.sync_thread = None
 
         # Set all the pending results to False, this will also
         # cause pending retries to be aborted
         for async_result in self._messageids_to_asyncresult.values():
             async_result.set(False)
+        gevent.wait(self.greenlets, timeout=self._logout_timeout)
```

Comment:
This would better be added to the `with gevent.Timeout` block below, otherwise we may end up waiting twice the allowed timeout.